### DISCUSSION TITLE
Is it possible to position dynamically created Handles using css flex positioning?

### DISCUSSION THREAD

**Discussion by tauren** [xyflow/xyflow#3852] [Q&A]
I'm dynamically adding and removing custom source Handles from a node. The custom node renders multiple divs using flex items with `space-around` positioning. The handles are rendered as a child of those flex items. This allows me to nicely position the handles using flexbox instead of creating custom JS positioning logic.

However, when I dynamically add or remove Handles, the existing connection endpoint locations (SVG endpoints) **sometimes** don't update. I've tried using `useUpdateNodeInternals`, but it doesn't seem to do anything. The only way I can get it to re-render is to change a connection to a target handle. 

https://github.com/xyflow/xyflow/assets/282084/86e65eb5-6474-44e3-8044-f8fd6c7f1eb4

Here's excerpts from the code to illustrate. Any idea why I'm getting invalid intermittent behavior?

```javascript
// Node
const { id } = node;
const { indexes } = node.data; // [3, 6, 9]

const connections = useHandleConnections({ type: 'source' });
const updateNodeInternals = useUpdateNodeInternals();

useEffect(() => {
  updateNodeInternals(id);
}, [connections, updateNodeInternals, id]);

return (
  <div className={`flex justify-around gap-1`}>
    {indexes.map((index, i) => (
      <CustomHandle key={i} index={index} />
    ))}
  </div>
);
```

```javascript
// Handle
export const CustomHandle = ({ index }) => {
  return (
    <div>
      <div>#{index}</div>
      <Handle id={`index-${index}`} type="source" position={Position.Bottom} />
    </div>
  );
};
```

```css
/* Standard tailwindcss classes */
.flex {
  display: flex;
}
.justify-around {
  justify-content: space-around;
}
.gap-1 {
  gap: 0.25rem;
}
```

**Comment by tauren** [comment]
I did want to mention I'm using *v12*, if that makes a difference. 

I also  noticed #3800 is having the same problem.

**Comment by bcakmakoglu** [comment]
I got this to work with a `useEffect` and `useUpdateNodeInternals`.

[Here's a sandbox for v11.](https://codesandbox.io/p/devbox/52f32g?file=%2Fpackage.json%3A12%2C6-12%2C15)

The same seems to work in v12 as well.

[Here's a sandbox for v12.](https://codesandbox.io/p/sandbox/reactflow-reactive-flow-forked-9pkcl6?file=%2Fsrc%2FFlow.tsx%3A16%2C1)

**Reply by Azim-Ahmed** [reply]
why codesandbox is not working @bcakmakoglu

**Reply by bcakmakoglu** [reply]
Recent codesandbox changes probably - they changed their pricing model and it messes up a lot of old sandboxes -.-

**Reply by Azim-Ahmed** [reply]
This is really sad, 
do you have any other ways to get the code?

**Reply by moklick** [reply]
The sandboxes are private by default now. 

However we need an alternative to codesandbox. It's really a mess.

**Reply by Bartel-C8** [reply]
@bcakmakoglu : Could you possibly just paste your solution here in code tags (```) in the meanwhile? Thanks!

**Reply by tauren** [reply]
I'm returning to this issue after a long time away from the project. I missed the code sample that you provided before codesandbox locked things down. Can you paste the solution here or in a gist?

**Reply by bcakmakoglu** [reply]
[Been a while but this sort of solution should work.](https://codesandbox.io/p/sandbox/nice-kowalevski-29f3cj?file=%2Fsrc%2Fnodes%2FPositionLoggerNode.tsx%3A7%2C17-7%2C35)
